librz/debug/p/debug_winkd.c

```diff
@@ -89,13 +116,61 @@ static int rz_debug_winkd_attach(RzDebug *dbg, int pid) {
 	kdctx = (KdCtx *)desc->data;
 
 	// Handshake
-	if (!winkd_sync(kdctx)) {
+	int ret = winkd_sync(kdctx);
+	if (ret < 0) {
 		RZ_LOG_ERROR("Could not connect to winkd\n");
 		return false;
+	} else if (!ret) {
+		RZ_LOG_VERBOSE("Already synced\n");
+		return true;
 	}
 	if (!winkd_read_ver(kdctx)) {
 		return false;
 	}
+
+	// Load PDB for kernel
+	WindModule *m, *mod = NULL;
+	RzList *modules = NULL;
+	if (kdctx->kernel_module.addr) {
+		ut32 timestamp;
+		if (get_module_timestamp(kdctx->kernel_module.addr, &timestamp)) {
+			mod = &kdctx->kernel_module;
+			mod->timestamp = timestamp;
+		}
+	}
+	if (!mod && kdctx->windctx.PsLoadedModuleList) {
+		modules = winkd_list_modules(&kdctx->windctx);
+		RzListIter *it;
+		rz_list_foreach (modules, it, m) {
+			RZ_LOG_DEBUG("%" PFMT64x " %s\n", m->addr, m->name);
+			if (rz_str_endswith(m->name, "\\ntoskrnl.exe")) {
+				mod = m;
+				break;
+			}
+		}
+	}
+	if (!mod) {
+		RZ_LOG_ERROR("Failed to find ntoskrnl.exe module\n");
+		rz_list_free(modules);
+		return false;
+	}
+	char *exepath, *pdbpath;
+	if (!winkd_download_module_and_pdb(mod,
+		    dbg->corebind.cfgGet(dbg->corebind.core, "pdb.server"),
+		    dbg->corebind.cfgGet(dbg->corebind.core, "pdb.symstore"),
+		    &exepath, &pdbpath)) {
+		RZ_LOG_ERROR("Failed to download module and pdb\n");
+		rz_list_free(modules);
+		return false;
+	}
+	dbg->corebind.cmdf(dbg->corebind.core, "e bin.baddr=0x%" PFMT64x "", mod->addr, pdbpath);
```

Comment:
Can be this converted to the function? Like previously used `dbg->corebind.cfgGet()` but `dbg->corebind.cfgSet()`?

Reply:
`pdbpath` is redundant here, isn't it?